src/App.tsx

```diff
@@ -15,48 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Name = 'name',
+  Length = 'length',
+}
+
+type SortType = SortBy | '';
+
+function sortGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+): string[] {
+  const copyOfGoods = [...goods];
+
+  copyOfGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortBy.Name:
+        return good1.localeCompare(good2);
+      case SortBy.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
```

Comment:
It would be great to move switch case out of the sort callback so you don't need to extra call sort method if it's a default case